src/Products/ZCatalog/tests/test_zcatalog.py

```diff
@@ -307,6 +314,16 @@ def testSearchNot(self):
     # manage_setProgress
     # _getProgressThreshold
 
+    def test_catalogView(self):
+        catalog = makerequest(self._catalog)
+        # hack `getPhysicalPath` to avoid problem with the catalog plan
+        catalog.getPhysicalPath = None
+        # provide `ZopeVersion`
+        catalog.ZopeVersion = lambda *arg, **kw: 4
+        vr = catalog.manage_catalogView()
+        self.assertTrue("There are no objects in the Catalog." not in vr,
```

Comment:
```suggestion
        self.assertNotIn("There are no objects in the Catalog.", vr,
```

Maybe an assertion what's actually in `vr` would be helpful as asserting something is not there is a relatively weak assertion. There are so many reasons something might not be there.

Reply:
The test directly verifies the faulty behavior introduced by changing the way empty searches are handled ("all objects" -> "no objects"). It caused the message `There are no objects in the Catalog` even though this was false. The test checks that this faulty behavior is gone. Testing that the template actually produces the desired output requires much more machinery. I checked this aspect visually.